src/woo_publications/publications/tests/test_document_api_endpoints.py

```diff
@@ -1217,49 +1217,446 @@ def test_partial_update_document_eigenaar(self):
+        published_revoked_document = DocumentFactory.create(
+            publicatiestatus=PublicationStatusOptions.published,
+            publicatie=revoked_publicatie,
+        )
+
+        published_concept_document_url = reverse(
+            "api:document-detail",
+            kwargs={"uuid": str(published_concept_document.uuid)},
+        )
+        published_published_document_url = reverse(
+            "api:document-detail",
+            kwargs={"uuid": str(published_published_document.uuid)},
+        )
+        published_revoked_document_url = reverse(
+            "api:document-detail",
+            kwargs={"uuid": str(published_revoked_document.uuid)},
+        )
+
+        with self.subTest("published document, concept publication -> concept"):
+            response = self.client.patch(
+                published_concept_document_url,
+                data={"publicatiestatus": PublicationStatusOptions.concept},
+                headers=AUDIT_HEADERS,
+            )
+
+            self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+            data = response.json()
+            self.assertEqual(
+                data["publicatiestatus"],
+                [
+                    _("You cannot set the status to {concept}.").format(
+                        concept=PublicationStatusOptions.concept.label.lower()
+                    )
+                ],
+            )
+
+        with self.subTest("published document, concept publication -> concept"):
+            response = self.client.patch(
+                published_concept_document_url,
+                data={"publicatiestatus": PublicationStatusOptions.concept},
+                headers=AUDIT_HEADERS,
+            )
+
+            self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+            data = response.json()
+            self.assertEqual(
+                data["publicatiestatus"],
+                [
+                    _("You cannot set the status to {concept}.").format(
+                        concept=PublicationStatusOptions.concept.label.lower()
+                    )
+                ],
+            )
+
+        with self.subTest("published document, concept publication -> published"):
+            response = self.client.patch(
+                published_concept_document_url,
+                data={"publicatiestatus": PublicationStatusOptions.published},
+                headers=AUDIT_HEADERS,
+            )
+
+            self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+            data = response.json()
+            self.assertEqual(
+                data["publicatiestatus"],
+                [
+                    _(
+                        "The given publicatiestatus isn't compatible with the "
+                        "publicatiestatus from the linked publication."
+                    )
+                ],
+            )
+
+        with self.subTest("published document, concept publication -> revoked"):
```

Comment:
this test code contains many state combinations that shouldn't be possible to begin with. A published document cannot occur within a concept publication when you look at Marco's state diagram. These tests are flawed.